src/core/ddsi/src/q_plist.c

```diff
@@ -310,6 +339,21 @@ static dds_return_t fini_locator (void * __restrict dst, size_t * __restrict dst
   return 0;
 }
 
+static const enum pserop* goto_sequence_end (const enum pserop * __restrict desc, const enum pserop * const desc_end)
```

Comment:
I'm not sold on this one (yet): the loops I put in to scan for the end of the (then still non-nested) sequence were never the prettiest thing. Perhaps you could consider changing it to, e.g., ``pserop_advance`` that returns ``desc+1`` if ``*desc==XQ`` and otherwise does what you do here; and then use that throughout instead of the ``desc++`` it uses now in all the loops?

(Do let me know if this description makes no sense!)